tests/pytests/test_coordinator.py

```diff
@@ -185,3 +183,33 @@ def test_mod_6287(env):
     # Send another command to make sure that the coordinator is healthy
     res = env.cmd('FT.AGGREGATE', 'idx', '*', 'LIMIT', '0', str(n_docs))
     env.assertEqual(len(res)-1, n_docs)
+
+def test_single_shard_optimization():
+    env = Env(shardsCount=1) # Either standalone or cluster with 1 shard
```

Comment:
How does the test verify that it bypass the coordinator? The test would have pass even without the changes that made on the PR no?

Reply:
I attempted to use commands that have a different reply on standalone and cluster configuration, but yes most of the time there is no difference (by design). We can only see the coverage and that there are no leaks when skipping the coordinator code. We can wait for another PR of mine that aligns the `FT.PROFILE` reply structure and we could see there that the coordinator wasn't used

Reply:
Maybe we can at least verify (inside the test) that `NumShards` is 1, using some debug command maybe?